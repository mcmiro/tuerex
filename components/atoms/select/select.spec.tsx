import * as React from "react";
import {screen, render} from "@testing-library/react";
import Select from "./";

describe("Test Select component", () => {
	it("Should pass when name, size, defaultValue, option and onSelect props added", () => {
		const optionsMock = ["test1", "test2"];
		const onSelectMock = jest.fn();

		render(<Select name="test" size="sm" options={optionsMock} onSelect={onSelectMock} />);

		const select = screen.getByRole("combobox");
		expect(select).toBeDefined();
		expect(select).toMatchSnapshot();
	});

	it("Should pass when option its not selected", () => {
		const optionsMock = ["test1", "test2"];
		const onSelectMock = jest.fn();

		render(<Select name="test" size="sm" options={optionsMock} onSelect={onSelectMock} />);
		const select = screen.getByRole("combobox") as HTMLElement;
		expect((select as HTMLOptionElement).selected).toBe(undefined);
	});
});
