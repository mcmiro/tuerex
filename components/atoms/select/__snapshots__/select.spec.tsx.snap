// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Select component Should pass when name, size, defaultValue, option and onSelect props added 1`] = `
<select
  class="w-full border rounded-sm py-[9px] px-[16px]"
  name="test"
>
  <option
    value="test1"
  >
    test1
  </option>
  <option
    value="test2"
  >
    test2
  </option>
</select>
`;
